### PR TITLE
feat: adds chip

### PR DESCRIPTION
**Pull request by stutirao**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/atoms/_chip/index.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import Icon from '@/components/atoms/icon';
+export type Type = 'Action' | 'Selection' | 'Input';
```
Comment (riyalohia) [inline core/components/atoms/_chip/index.tsx]: Type will not be used in GenericChip.

File: core/components/atoms/_chip/index.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  clearbutton?: boolean;
+  disabled?: boolean;
+  selected?: boolean;
+  className?: string;
+  onClose?: (name?:any) => void;
+  onClick?: (name?: any) => void;
+}
+export const GenericChip = (props: GenericChipProps) => {
+  const {
+    label = 'Selection',
```
Comment (riyalohia) [inline core/components/atoms/_chip/index.tsx]: Remove default label

File: core/components/atoms/_chip/index.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    icon,
+    clearbutton,
+    disabled,
+    className,
+    selected,
+    onClose,
+    onClick,
+    ...rest
```
Comment (riyalohia) [inline core/components/atoms/_chip/index.tsx]: No need to add rest

File: core/components/atoms/_chip/index.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        {icon && (
+          <div className={iconClass('left')}>
+            <Icon
+              name={icon}
+              size={16}
+              appearance={(disabled ? 'disabled' : (selected ? 'info' : 'default'))}
+            />
+          </div>
+        )}
+        <div className="Chiplabel">
```
Comment (riyalohia) [inline core/components/atoms/_chip/index.tsx]: Replace Chiplabel class with Text component

File: core/components/atoms/_chip/index.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          {label}
+        </div>
+      </div>
+      {(
+        clearbutton &&
+        (
+          <div onClick={onCloseHandler}>
+            <div className={iconClass('right')}>
```
Comment (riyalohia) [inline core/components/atoms/_chip/index.tsx]: Add onClick event handler on this div

File: core/components/atoms/_chip/index.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    [`Chip-icon--${align}`]: align,
+  });
+  const onCloseHandler = () => {
+    if (onClose) onClose(name);
+  };
+  const onClickHandler = () => {
+    if (onClick) onClick(name);
+  };
+  return (
+    <div className={classes} onClick={onClickHandler} {...rest}>
```
Comment (riyalohia) [inline core/components/atoms/_chip/index.tsx]: Replace classes with prop className

File: core/components/atoms/chip/Chip.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  clearbutton?: boolean;
+  disabled?: boolean;
+  selected?: boolean;
+  type?: Type;
+  onClose?: (name?: any) => void;
+  onClick?: (name?: any) => void;
+}
+export const Chip = (props: ChipProps) => {
+  const {
+    label = 'Selection',
```
Comment (riyalohia) [inline core/components/atoms/chip/Chip.tsx]: Remove default label

File: core/components/atoms/chip/_stories_/index.story.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { select, boolean, text } from '@storybook/addon-knobs';
+import Chip from '../Chip';
+export type Type = 'Action' | 'Selection' | 'Input';
```
Comment (riyalohia) [inline core/components/atoms/chip/_stories_/index.story.tsx]: import Type from Chip.tsx

File: core/components/atoms/chip/Chip.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    clearbutton,
+    type,
+    disabled,
+    selected,
+    onClose,
+    onClick,
+
+  } = props;
+
+  const onCloseHandler = () => {
```
Comment (riyalohia) [inline core/components/atoms/chip/Chip.tsx:30]: Add name in props

File: core/components/atoms/chip/_stories_/index.story.tsx [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+  };
+  const onClickHandler = (name?: any) => {
+    return action(`onClick: ${name}`)();
+  };
+
+  if (!open) return null;
+
+  return (
+    <div>
+      {
```
Comment (riyalohia) [inline core/components/atoms/chip/_stories_/index.story.tsx]: Remove braces

File: core/components/atoms/chip/_stories_/variants/Action.story.tsx [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+          label={label}
+          clearbutton={true}
+          disabled={false}
+          type="Action"
+          onClose={onCloseHandler}
+          onClick={onClickHandler}
+          selected={false}
+        />
+      </div>
+      <div style={innerStyle}>
```
Comment (riyalohia) [inline core/components/atoms/chip/_stories_/variants/Action.story.tsx]: Use a loop for boolean prop disabled.

File: css/src/components/chip.css [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+    font-family: Nunito Sans;
+    text-align: left;
+    line-height: var(--font-height);
+  }
+  
+  .Chip-icon--left {
+    padding-right: var( --spacing-m);
+    display: flex;
+    align-items: center;
+    color: var(--night);
```
Comment (riyalohia) [inline css/src/components/chip.css]: No need to add color

File: css/src/components/chip.css [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+  }
+  
+  .Chip-icon--right {
+    padding-left: var( --spacing-m);
+    display: flex;
+    align-items: center;
+    color: var(--night-light);
```
Comment (riyalohia) [inline css/src/components/chip.css]: No need to add color as this is an Icon component.

File: css/src/components/chip.css [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+  }
+  
+  .Chip--Action {
+    background: var( --shadow-0);
+    border-color: var(--stone);
+    font-weight: var( --font-weight-normal);
```
Comment (riyalohia) [inline css/src/components/chip.css]: Remove font-weight

File: css/src/components/chip.css [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+  .Chip--disabled--Action {
+    background: var(--shadow-0);
+    color: var(--night-lightest);
+    border-color: var(--stone-light);
+  }
+  
+  .Chip--Selection {
+    background: var( --shadow-0);
+    border-color: var(--stone);
+    font-weight: var( --font-weight-normal);
```
Comment (riyalohia) [inline css/src/components/chip.css]: Remove font-weight and cursor

File: css/src/components/chip.css [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+  
+  .Chip--Action:active {
+    background: var(--stone-dark);
+    cursor: pointer;
+  }
+  
+  .Chip--disabled--Action {
+    background: var(--shadow-0);
+    color: var(--night-lightest);
+    border-color: var(--stone-light);
```
Comment (riyalohia) [inline css/src/components/chip.css:54]: Add default cursor for disabled state and remove color

File: css/src/components/chip.css [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+  
+  .Chip--Selection:active {
+    background: var(--stone-dark);
+    cursor: pointer;
+  }
+  
+  .Chip--Selection--selected {
+    background: var(--jal-lightest);
+    border-color: var(--jal-lighter);
+    color: var(--jal);
```
Comment (riyalohia) [inline css/src/components/chip.css]: Remove color, font-weight and cursor

File: css/src/components/chip.css [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+    font-weight: var( --font-weight-medium);
+    cursor: pointer;
+  }
+  
+  .Chip--disabled--Selection {
+    background: var( --shadow-0);
+    color: var(--night-lightest);
```
Comment (riyalohia) [inline css/src/components/chip.css]: Remove color

File: core/components/atoms/chip/Chip.tsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+    if (onClose) onClose(name);
+  };
+  const onClickHandler = () => {
+    if (onClick) onClick(name);
+  };
+
+  const chipClass = classNames({
+    [`Chip--disabled--${type}`]: disabled,
```
Comment (riyalohia) [inline core/components/atoms/chip/Chip.tsx]: class should be Chip-${type}--disabled

File: core/components/atoms/chip/Chip.tsx [hunk 20]
```diff
@@ -0,0 +1,61 @@
...
+    [`Chip--${type}`]: type && !disabled,
+    Chip: true,
+    [`Chip--${type}--selected`]: selected && !disabled,
```
Comment (riyalohia) [inline core/components/atoms/chip/Chip.tsx]: Class should be chip-${type}--selected

**Review by riyalohia — CHANGES_REQUESTED** [review]

File: css/src/components/chip.css (hunk 14 above)
Comment (aditya-kumawat) [reply css/src/components/chip.css]: Cursor is required for Action chips.

File: css/src/components/chip.css (hunk 16 above)
Comment (aditya-kumawat) [reply css/src/components/chip.css:54]: Also follow camelCase naming convention for modifiers too.

File: core/components/atoms/chip/_stories_/index.story.tsx [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+    );
+    updateKnob('label', 'Selection');
+  }
+  if (type === 'action') {
+    updateKnob('label', 'Action');
+  }
+  if (type === 'input') {
+    updateKnob('label', 'Input');
+  }
+  const open = boolean('open', true);
```
Comment (satyamyadav) [inline core/components/atoms/chip/_stories_/index.story.tsx]: @stutirao  Please remove `open` from knobs as it is not a prop for the component it creates confusion.

**Review by satyamyadav — CHANGES_REQUESTED** [review]
@stutirao  Please remove the `WIP` from the pull request title.

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**